src/lib.rs

```diff
@@ -53,7 +53,7 @@ use traits::{
 };
 
 /// A type that holds public parameters of Nova
-#[derive(Serialize, Deserialize)]
+#[derive(Serialize, Deserialize, Clone)]
```

Comment:
Why do we need Clone on public parameters? This type is always passed by reference, so clone seems unnecessary.